internal/tasks/fleet_rollout.go

```
@@ -0,0 +1,87 @@
+package tasks
+
+import (
+	"context"
+
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/flightctl/flightctl/pkg/reqid"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+func FleetRollouts(taskManager TaskManager) {
+	reqid.OverridePrefix("fleet-rollout")
+
+	for {
+		select {
+		case fleetRef := <-taskManager.channels[FleetTemplateRollout]:
+			rolloutFleet(taskManager.log, taskManager.store.Fleet(), taskManager.store.Device(), fleetRef)
+		case <-taskManager.ctx.Done():
+			return
+		}
+	}
+}
+
+func rolloutFleet(baseLog logrus.FieldLogger, fleetStore store.Fleet, devStore store.Device, fleetRef ResourceReference) {
+	requestID := reqid.NextRequestID()
+	ctx := context.WithValue(context.Background(), middleware.RequestIDKey, requestID)
+	log := log.WithReqIDFromCtx(ctx, baseLog)
+
+	log.Infof("Rolling out fleet %s/%s", fleetRef.OrgID, fleetRef.Name)
+
+	fleet, err := fleetStore.Get(ctx, fleetRef.OrgID, fleetRef.Name)
+	if err != nil {
+		log.Errorf("failed to get fleet: %v", err)
+		return
```

lets return the errors and log above.
```
err := rolloutFleet(taskManager.log, taskManager.store.Fleet(), taskManager.store.Device(), fleetRef)
if err != nil {
    log.Errorf("rollout failed: %v", err)
}
```